packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -52,15 +52,22 @@ export default function useMergeRefs( refs ) {
 
 	// There should be no dependencies so that `callback` is only called when
 	// the node changes.
-	return useCallback( ( value ) => {
+	const rootCallback = useCallback( ( value ) => {
 		// Update the element so it can be used when calling ref callbacks on a
 		// dependency change.
 		element.current = value;
 		didElementChange.current = true;
 
 		// When an element changes, the current ref callback should be called
 		// with the new element and the previous one with `null`.
-		const refsToUpdate = value ? currentRefs.current : previousRefs.current;
+		let refsToUpdate;
+		if ( value ) {
+			refsToUpdate = currentRefs.current;
+			// Makes this callback mimic a ref
+			rootCallback.current = value;
```

Comment:
It seems this PR might be fixing the consequence and not the root of the issue. Where is the `.current` value being used in useBlockProps?

Reply:
Created an alternative fix here #29288

Reply:
Nowhere within `useBlockProps`. From what I can tell the root of the issue is a breaking change in https://github.com/WordPress/gutenberg/pull/28917#discussion_r579859108, this just patches so the previous expectation can be met. This same type of patch could probably be done from within `useBlockProps` instead if that is better in any way. I don't have a problem with the breaking change, but just figured it might be nice to avoid it if possible since third-party blocks could break.

Reply:
Indeed, I'd like to understand how impactful is the breaking change in other code aside Core code. For me, code relying on the fact that "ref" is an object is a code that need to be fixed anyway (core or third-party), so depending on the impact, I might be ok with this breaking change.